STYLE: Add SampleData.downloadSample

jcfr opened the pull request:
This is a follow of https://github.com/Slicer/Slicer/pull/1062

Other possible improvement:
* decouple downloading and loading by having `loadFromURL` and `downloadFromURL`
* rename `downloadSample` into `loadSample`. Since this may break existing code .. may be we should keep this for Slicer 5.0 ?

lassoan commented on Applications/SlicerApp/Testing/Python/SlicerBoundsTest.py:
```diff
@@ -71,8 +71,8 @@ def test_Volume(self):
-    from SampleData import SampleDataLogic
-    volumeNode = SampleDataLogic().downloadAbdominalCTVolume()
+    import SampleData
+    volumeNode = SampleData.downloadSample('CTA abdomen\n(Panoramix)')[0]
```
Yes, it would be nicer to separate name and description.

lassoan commented on Modules/Loadable/Markups/Testing/Python/NeurosurgicalPlanningTutorialMarkupsSelfTest.py:
```diff
@@ -143,8 +143,8 @@ def run(self):
-    from SampleData import SampleDataLogic
-    baselineVolume = SampleDataLogic().downloadWhiteMatterExplorationBaselineVolume()
+    import SampleData
+    baselineVolume = SampleData.downloadSample('BaselineVolume')[0]
```
We can certainly rename it for Slicer5. There are more and more backward-incompatible changes that we queue up. We should release Slicer-4.10.1 as soon as possible.

lassoan commented on Applications/SlicerApp/Testing/Python/BRAINSFitRigidRegistrationCrashIssue4139.py:
```diff
@@ -103,12 +103,12 @@ def test_BRAINSFitRigidRegistrationCrashIssue4139(self):
+    import SampleData
 
-    fixed = SampleDataLogic().downloadMRBrainTumor1()
+    fixed = SampleData.downloadSample('MRBrainTumor1')[0]
```
The [0] at the end does not look nice. What do you think about having `downloadSamples` - returning all files in the sample; and `def downloadSample(*arg): downloadSamples[0]` returning the first sample?

jcfr replied:
Agreed.

jcfr commented:
Closing. All comment have been addressed:
* added downloadSample and downloadSamples. This broke only one extension, associated PR is https://github.com/SlicerRt/Sequences/pull/65 and has been integrated
* added support for `sampleDescription`

See [r27648](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=27648), [r27649](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=27649) and [r27650](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=27650)

lassoan commented:
Perfect, thank you!
